presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -75,13 +81,16 @@ public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
             // FilePath is a unique identifier for a file, however it can be a long string
             // hence using md5 hash of the file path as the identifier in the cache.
             // We don't set fileId because fileId is Alluxio specific
-            FileInfo info = new FileInfo().setFileIdentifier(md5().hashString(path.toString(), UTF_8).toString())
+            FileInfo info = new FileInfo()
+                    .setLastModificationTimeMs(hiveFileContext.getModificationTime())
                     .setPath(path.toString())
                     .setFolder(false)
                     .setLength(hiveFileContext.getFileSize().get());
-            // URIStatus is the mechanism to pass the hiveFileContext to the source filesystem
-            AlluxioURIStatus alluxioURIStatus = new AlluxioURIStatus(info, hiveFileContext);
-            FSDataInputStream cachingInputStream = localCacheFileSystem.open(alluxioURIStatus, BUFFER_SIZE);
+            String cacheIdentifier = md5().hashString(path.toString(), UTF_8).toString();
+            // CacheContext is the mechanism to pass the cache related context to the source filesystem
+            CacheContext cacheContext = PrestoCacheContext.build(cacheIdentifier, hiveFileContext, cacheQuotaEnabled);
+            URIStatus uriStatus = new URIStatus(info, cacheContext);
+            FSDataInputStream cachingInputStream = localCacheFileSystem.open(uriStatus, BUFFER_SIZE);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code contains a resource that might not have closed properly. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource: *cachingInputStream*. The resource is referenced at line 95. The resource is returned at line 98. There are other execution paths that don&#39;t close the resource or return it, for example, when ExtendedFileSystem.openFile throws an exception, and an if condition at line 94 evaluates to true. To prevent this resource leak, close *cachingInputStream* along these other paths before you exit this method.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).